### PR TITLE
fix(glimmer): {{else}}{{#if}} into {{else if}} merging

### PR DESCRIPTION
**Pull request by dcyriller**
## Description
The first commit highlights the issue. The second fixes it.

The following code should be left untouched:
```
{{#if a}}
  a
{{else}}
  {{#if c}}
    c
  {{/if}}
  e
{{/if}}
```
Instead, it was turned into:
```
{{#if a}}
  a
{{else if c}}
  c
e
{{/if}}
```

## Checks
- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

## Playground
**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/glimmer/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -872,6 +872,16 @@ printWidth: 80
...
+
+{{#if a}}
+  a
+{{else}}
+  {{#if c}}
+    c
+  {{/if}}
+  e
+{{/if}}
+
```
Comment (evilebottnawi) [inline tests/glimmer/__snapshots__/jsfmt.spec.js.snap]: Please add tests where in `if`/`else` more then one character, like:
```
{{#if a}}
  a
  b
  c
{{else}}
  {{#if c}}
    a
    b
    c
  {{/if}}
  a
  b
  c
{{/if}}
```

Comment (dcyriller) [reply tests/glimmer/__snapshots__/jsfmt.spec.js.snap]: This is done 👍

**Comment by dcyriller** [comment]
@evilebottnawi done, sorry for that

**Comment by dcyriller** [comment]
@evilebottnawi `CHANGELOG.unreleased.md` is up to date.

**Comment by evilebottnawi** [comment]
@dcyriller Thanks, can you fix CI?

**Comment by dcyriller** [comment]
@evilebottnawi CI is now 🍏

I see that you squash the commits, so I don't bother to remove the last one (that was empty).

**Pull request merged**